app/forms/choose_role_form.rb

```diff
@@ -47,21 +44,50 @@ def redirect_path(helpers:)
   end
 
   def only_one_role
-    return false unless user.user_roles.count == 1
+    return false unless sanitized_user_roles.count == 1
 
-    self.role = user.user_roles.first
+    self.role = sanitized_user_roles.first
     true
   end
 
   def has_no_role
-    return false unless user.user_roles.count.zero?
+    return false unless sanitized_user_roles.empty?
 
     self.role = "no_role"
     true
   end
 
 private
 
+  def rejected_roles
+    %w[induction_coordinator early_career_teacher mentor npq_applicant npq_participant lead_provider].freeze
+  end
+
+  def sanitized_user_roles
+    roles = user_roles.reject { |role| rejected_roles.include?(role) }
+
+    if sit_role?
+      roles << "induction_coordinator"
+    elsif teacher_role?
+      roles << "teacher"
+    end
+    roles
+  end
```

Comment:
Can we move this to `User.user_roles`? that way it can be the source of truth.
We won't need to sanitize in this form as `USER_ROLES` already filters out roles we don't care about.

Reply:
But _we_ care about them?  If we remove them at source then there is no point in adding them in the first place. A bit confused I thought we'd gone over this already?